website/static/js/keen.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+var keen = require('keen-js');
+var oop = require('js/oop');
+var $ = require('jquery');
+var uuid = require('uuid');
+
+var KeenTracker = oop.defclass({
+    constructor: function(keenProjectId, keenWriteKey) {
+        this.keenClient = new keen({
+            projectId: keenProjectId,
+            writeKey: keenWriteKey
+        });
+        this.init();
+    },
+
+    createOrUpdateKeenSession: function() {
+        var date = new Date();
+        var expiresInMinutes = 25;
+        var expDate = date.setTime(date.getTime() + (expiresInMinutes*60*1000));
+        if(!$.cookie('keenSessionId')){
+            $.cookie('keenSessionId', uuid.v1(), {expires: expDate, path: '/'});
+        } else {
+            var sessionId = $.cookie('keenSessionId');
+            $.cookie('keenSessionId', sessionId, {expires: expDate, path: '/'});
+        }
+    },
+
+    getOrCreateKeenId: function() {
+        if(!$.cookie('keenUserId')){
+            $.cookie('keenUserId', uuid.v1(), {expires: 365, path: '/'});
+        }
+
+        return $.cookie('keenUserId');
+    },
+
+    trackPageView: function(){
+        this.createOrUpdateKeenSession();
+        var ctx = window.contextVars;
```

Comment:
`contextVars` should not be used in re-usable modules. Instead, any data this class depends on should be passed in the constructor.